app/javascript/gobierto_visualizations/modules/costs_controller.js

```diff
@@ -150,22 +144,18 @@ export class CostsController {
       "costdirfin"
     ];
 
-    const population = ["128291", "129661"];
+    const population = ["128291", "129661", "129120"];
 
     let yearsCosts = [...new Set(rawData.map(item => item.any_))];
 
-    for (let index = 0; index < rawData.length; index++) {
-      let d = rawData[index];
-
-      const ix = yearsCosts.findIndex(x => x === d["any_"])
-      d.population = population[ix]
-
-      for (let amounts = 0; amounts < amountStrings.length; amounts++) {
-        d[amountStrings[amounts]] = convertStringToNumbers(
-          d[amountStrings[amounts]]
+    rawData.forEach((item, index) => {
+      rawData[index].population = population[yearsCosts.findIndex(year => year === rawData[index]["any_"])]
+      for (let amounts of amountStrings) {
+        rawData[index][amounts] = convertStringToNumbers(
+          rawData[index][amounts]
         );
       }
-    }
+    })
```

Comment:
Now you begin, try to reformule this block as the functional programming recommends you:
- Don't mutate objects (therefore, `const data = rawData.map(...)`)
- Create small functions to leave crystal clear what're you doing: `const setPopulation = () => {}`, `const toNumber = () => {}`, etc.

Hit me with any doubt

Reply:
Done!